django_rq/decorators.py

```diff
@@ -30,13 +34,17 @@ def job(func_or_queue, connection=None, *args, **kwargs):
                 connection = queue.connection
         except KeyError:
             pass
+    else:
+        if connection is None:
+            connection = queue.connection
 
     RQ = getattr(settings, 'RQ', {})
     default_result_ttl = RQ.get('DEFAULT_RESULT_TTL')
     if default_result_ttl is not None:
         kwargs.setdefault('result_ttl', default_result_ttl)
 
-    decorator = _rq_job(queue, connection=connection, *args, **kwargs)
+    kwargs['connection'] = connection
```

Comment:
This is just for mypy (https://github.com/python/mypy/issues/6799) and rather than adding a type ignore I figured I'd update the kwargs, since it would never contain the connection based on the function's definition